### ISSUE TITLE
Definite integral returns an answer with indefinite integrals

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#7130]
```
In [1]: var("i L a b")
Out[1]: (i, L, a, b)

In [2]: integrand =  cos(pi*i*x/L)**2/(a + b*x)

In [3]: integrand = integrand.rewrite(exp)

In [4]: integrate(integrand, (x, 0, L))
Out[4]:
           ⌠
  ⌠        ⎮    4⋅ⅈ⋅π⋅i      2⋅ⅈ⋅π⋅i
  ⎮ 1      ⎮   ℯ        + 2⋅ℯ        + 1
- ⎮ ─ dx + ⎮ ───────────────────────────── dx
  ⎮ a      ⎮        2⋅ⅈ⋅π⋅i        2⋅ⅈ⋅π⋅i
  ⌡        ⎮ 4⋅L⋅b⋅ℯ        + 4⋅a⋅ℯ
           ⌡

The answer is obviously wrong (it should not depend on x).

Also, I don't know if this is related, but integrate seems to be quite inefficient. I put a print statement at the top of integrate() to print the args, and I got

In [4]: integrate(integrand, (x, 0, L))
((exp(I*pi*i*x/L)/2 + exp(-I*pi*i*x/L)/2)**2/(a + b*x), (x, 0, L))
(4/(4*a + 4*b*x) - 1/(2*a + 2*b*x), x)
(1/(2*a + 2*b*x), x)
(1/(4*a + 4*b*x), x)
(1/(4*a + 4*b*x), x)
(0, x)
(1/(4*a + 4*b*x), x)
(1/(4*a + 4*b*x), x)
(1/(4*a + 4*b*x), x)
Out[4]:
           ⌠
  ⌠        ⎮    4⋅ⅈ⋅π⋅i      2⋅ⅈ⋅π⋅i
  ⎮ 1      ⎮   ℯ        + 2⋅ℯ        + 1
- ⎮ ─ dx + ⎮ ───────────────────────────── dx
  ⎮ a      ⎮        2⋅ⅈ⋅π⋅i        2⋅ⅈ⋅π⋅i
  ⌡        ⎮ 4⋅L⋅b⋅ℯ        + 4⋅a⋅ℯ
           ⌡

In other words, it is computing that same integral (1/(4*a + 4*b*x)) six times.
```
Original issue for #7130: http://code.google.com/p/sympy/issues/detail?id=4031
Original author: https://code.google.com/u/asmeurer@gmail.com/

**Comment by smichr** [comment]
The result no longer depends on x:

```python
>>> i=integrate(integrand, (x, 0, L))
>>> i.free_symbols
set([a, L, b, i])
```

So perhaps this can be closed with a test.

**Comment by asmeurer** [comment]
Yeah, I seem to remember a PR that fixed Integral.subs and this was probably fixed by it. 

```
In [5]: integrate(integrand, (x, 0, L))
Out[5]:
  0                                    L                                    0                                    L
  ⌠                                    ⌠                                    ⌠                                    ⌠
  ⎮             2⋅ⅈ⋅π⋅i⋅x              ⎮             2⋅ⅈ⋅π⋅i⋅x              ⎮            4⋅ⅈ⋅π⋅i⋅x               ⎮            4⋅ⅈ⋅π⋅i⋅x
  ⎮             ─────────              ⎮             ─────────              ⎮            ─────────               ⎮            ─────────
  ⎮                 L                  ⎮                 L                  ⎮                L                   ⎮                L
  ⎮          2⋅ℯ                       ⎮          2⋅ℯ                       ⎮           ℯ                        ⎮           ℯ
  ⎮ ───────────────────────────── dx   ⎮ ───────────────────────────── dx   ⎮ ───────────────────────────── dx   ⎮ ──────────────────────
  ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅ⅈ⋅π⋅i⋅x      ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅ⅈ⋅π⋅i⋅x      ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅ⅈ⋅π⋅i⋅x      ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅
  ⎮    ─────────        ─────────      ⎮    ─────────        ─────────      ⎮    ─────────        ─────────      ⎮    ─────────        ──
  ⎮        L                L          ⎮        L                L          ⎮        L                L          ⎮        L
  ⎮ a⋅ℯ          + b⋅x⋅ℯ               ⎮ a⋅ℯ          + b⋅x⋅ℯ               ⎮ a⋅ℯ          + b⋅x⋅ℯ               ⎮ a⋅ℯ          + b⋅x⋅ℯ
  ⌡                                    ⌡                                    ⌡                                    ⌡

- ────────────────────────────────── + ────────────────────────────────── - ────────────────────────────────── + ────────────────────────
                  4                                    4                                    4                                    4




             0                                    L
             ⌠                                    ⌠
             ⎮               1                    ⎮               1
─────── dx   ⎮ ───────────────────────────── dx   ⎮ ───────────────────────────── dx
ⅈ⋅π⋅i⋅x      ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅ⅈ⋅π⋅i⋅x      ⎮    2⋅ⅈ⋅π⋅i⋅x        2⋅ⅈ⋅π⋅i⋅x
───────      ⎮    ─────────        ─────────      ⎮    ─────────        ─────────
  L          ⎮        L                L          ⎮        L                L
             ⎮ a⋅ℯ          + b⋅x⋅ℯ               ⎮ a⋅ℯ          + b⋅x⋅ℯ
             ⌡                                    ⌡

────────── - ────────────────────────────────── + ──────────────────────────────────
                             4                                    4
```

**Comment by Jawadh-Salih** [comment]
Hi. I would like  on this bug. So could you please help me with this. 
When I went through this. I saw like the final answer of an Integral gives a functions (depend on a varialble) instead of a constant value. Am I correct?

**Comment by asmeurer** [comment]
I believe the issue has already been fixed so only a test needs to be added for it.

**Comment by debugger22** [comment]
Test added in https://github.com/sympy/sympy/pull/9161
